actioncable/lib/action_cable.rb

```diff
@@ -37,6 +37,10 @@ module ActionCable
     },
     default_mount_path: '/cable'.freeze
   }
+  
+  # This protocol ought to be versioned separately
+  # from the rest of the Gem
+  PROTOCOLS = ["actioncable-protocol-1.0.0".freeze].freeze
```

Comment:
How about e.g. `actioncable-v17-json`? Tack on the transport encoding (which is always JSON now) and use a big chunky version that looks nothing like the AC version number since the only purpose is to break major compat (there are no "minor" changes).

Then the server can negotiate by parsing out the requested version and transfer encoding and picking the matching protocol handler for the conn.